src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt

```diff
@@ -90,11 +90,9 @@ class Personfakta(val datagrunnlag: Datagrunnlag) {
             periodefilter(lagInterval(Periode(it.fom, it.tom)), datohjelper.kontrollPeriodeForNorskAdresse())
 
     fun sjekkBrukersPostadresseOgBostedsadresseLandskode(): Boolean {
-        val harUtenlandskBostedsadresse = !harBrukerNorskBodstedsadresseInnenforSiste12Mnd()
-        val harUtenlandskPostadresse = datagrunnlag.personhistorikk.postadresser.isNotEmpty() &&
-                !harBrukerNorskPostadresseInnenforSiste12Mnd()
-
-        return !(harUtenlandskBostedsadresse || harUtenlandskPostadresse)
+        val harIngenPostadresse = datagrunnlag.personhistorikk.postadresser.isEmpty()
+        return harBrukerNorskPostadresseInnenforSiste12Mnd() || harIngenPostadresse &&
```

Comment:
Jeg er ikke stø nok til å vite om && eller || skal evalueres først, kan jeg anbefale et par paranteser..? :)

Reply:
Jeg antar at det skal leses som harBrukerNorskPostadresseInnenforSiste12Mnd || (harIngenPostadresse && harBrukerNorskBodstedsadresseInnenforSiste12Mnd) ? Så hvis bruker har norsk postadresse, og da bryr vi oss ikke om bostedsadressen, eller hvis bruker har norsk bostedsadresse og ingen postadresse, så er det greit. Ellers går det til uavklart? Jeg tenker vel kanskje at det er greit frem til Stians branch kommer inn, selv om jeg er litt usikker på om en norsk postadresse skal kunne "vinne" over en utenlandsk bostedsadresse. Jeg tror det er litt snillere enn Helle egentlig ønsker seg.

Reply:
Hmm. jeg tenker det leses slik: 

(harIngenPostadresse || harBrukerNorskPostadresseInnenforSiste12Mnd ) && harBrukerNorskBodstedsadresseInnenforSiste12Mnd

Og kan legge inn paranteser. Jeg tenker at hun ville fange opp alle med utenlandske postadresser? men mulig jeg misforstår. Kan lese en ekstra gang over confluence sidene 

 Det er skrevet tester på dette nå som gir ønsket resultat. Tenker dette bare er en lettere å lese det på også.

Reply:
I ditt forslag vil en person med norsk postadresse og utenlandsk bostedsadresse gi JA. Blir vell ikke helt riktig det?

Reply:
Nei, enig, det var derfor jeg skrev at det virket litt snillere enn det Helle ønsket seg. :)

Reply:
Du har snutt litt om på det da, tidligere sjekket vi om bruker ikke hadde utenlandsk adresse, nå sjekker du om bruker har norsk adresse. Det kan være at det er litt for snilt, hva hvis bruker har både norsk og utenlandsk?

Reply:
Men jeg misforstod åpenbart hvor jeg skulle tenke parantesene, så bra at du legger til eksplisitte paranteser! :)

Reply:
Litt usikker om jeg henger med, men den gamle metoden returnerer til slutt: 

!(harUtenlandskBostedsadresse || harUtenlandskPostadresse)

Den nye metoden returnerer en forenkling: 

return (harBrukerNorskPostadresseInnenforSiste12Mnd() || harIngenPostadresse) &&
                harBrukerNorskBodstedsadresseInnenforSiste12Mnd()

Hvor ingen postadresse tilsvarer norsk postadresse som Helle sa i møtet var egentlig det tryggeste scenario. 

Den gamle mente jeg hadde en feil med å ha med kodelinjen: 

val harUtenlandskPostadresse = datagrunnlag.personhistorikk.postadresser.isNotEmpty() &&

Der harUtenlandskPostadresse gir false om en person bosatt i norge uten postadresse vil gå til uavklart. 

Både norsk og utenlandsk vil gi UAVKLART (noe jeg mener er det hun ønsker å fange opp). Men mulig jeg har misforstått

Reply:
Tenker du at norsk bostedsadresse bør overstyre alt?

Reply:
Det er litt komplisert.. Jeg tenker at "har ikke utenlands adresse" og "har norsk adresse" ikke nødvendigvis er det samme. Men jeg tror nok kanskje at det er greit å kjøre på med dette, frem til Stians ekstra-adresser kommer inn.